src/app.js

```diff
@@ -49,54 +41,89 @@ export function app(props, container) {
     )
   }
 
-  function init(module, state, actions) {
-    if (module.init) {
-      callbacks.push(function() {
-        module.init(state, actions)
-      })
-    }
-
+  function init(module, state, actions, path) {
     assign(state, module.state)
-
-    initActions(state, actions, module.actions)
+    create(state, actions, module.actions, path)
 
     for (var i in module.modules) {
-      init(module.modules[i], (state[i] = {}), (actions[i] = {}))
+      init(
+        module.modules[i],
+        (state[i] = {}),
+        (actions[i] = {}),
+        path.concat(i)
+      )
     }
-  }
 
-  function initActions(state, actions, source) {
-    Object.keys(source || {}).map(function(i) {
-      if (typeof source[i] === "function") {
-        actions[i] = function(data) {
-          return typeof (data = source[i](state, actions, data)) === "function"
-            ? data(update)
-            : update(data)
+    function create(state, actions, from, path) {
+      Object.keys(from || {}).map(function(i) {
+        if (typeof from[i] === "function") {
+          actions[i] = function(data) {
+            if (
+              typeof (state = from[i](
+                getObject(path, globalState),
+                actions
+              )) === "function"
+            ) {
+              state = state(data)
+            }
+
+            if (
+              state &&
+              !state.then &&
+              state !== (data = getObject(path, globalState))
```

Comment:
No more getters? :(

Reply:
🤕 No more getters.

Reply:
You can make a getter that return a state slice without triggering a render isn't it ?

Reply:
@Swizz Yes.